tests/Http/Controllers/CelebTest.php

```diff
@@ -2,175 +2,414 @@
 {
     use RefreshDatabase;
 
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Carbon::setTestNow(now());
+    }
+
     /** @test */
     public function admin_can_see_celebs_create_view()
     {
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
-        ]);
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
             ->getJson('celebs/create')
             ->assertOk();
     }
 
     /** @test */
-    public function normal_user_cannot_see_celebs_create_view()
+    public function user_cannot_see_celebs_create_view()
     {
-        $user = \App\Models\User::factory()->create([
+        $user = User::factory()->create([
             'is_admin' => false
         ]);
 
         $this->actingAs($user)
             ->getJson('celebs/create')
-            ->assertStatus(403);
+            ->assertRedirect();
     }
 
     /** @test */
     public function guest_cannot_see_celebs_create_view()
     {
         $this->getJson('celebs/create')
-            ->assertStatus(403);
+            ->assertRedirect();
     }
 
     /** @test */
-    public function admin_can_access_celebs_store()
+    public function admin_can_create_a_new_celeb()
     {
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->post('celebs', [
+                'name' => 'John Farnaby',
+                'date_of_birth' => Carbon::now()->subYears(30),
+                'photo' => 'https://generic/photo_700x600'
+            ])->assertSessionHasNoErrors();
+
+        $this->assertDatabaseHas('celebs', [
+            'name' => 'John Farnaby',
+            'date_of_birth' => Carbon::now()->subYears(30),
+            'photo' => 'https://generic/photo_700x600'
         ]);
+    }
+
+    /** @test */
+    public function validation_error_when_celeb_has_one_invalid_data()
+    {
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
-            ->postJson('celebs')
-            ->assertOk();
+            ->post('celebs', [
+                'name' => 'James Weatherby',
+                'date_of_birth' => Carbon::now()->subYears(30),
+                'photo' => null
+            ])->assertSessionHasErrors([
+                'photo' => 'The photo field is required.'
+            ]);
+
+        $this->assertDatabaseMissing('celebs', [
+            'name' => 'James Weatherby',
+            'date_of_birth' => Carbon::now()->subYears(30),
+            'photo' => null
+        ]);
+    }
+
+    /** @test */
+    public function validation_errors_when_celeb_has_all_invalid_data()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->post('celebs', [
+                'name' => null,
+                'date_of_birth' => null,
+                'photo' => null
```

Comment:
This one you can also simplify by just posting an empty array 😄 

And in that case, we do not need to check if the DB has the object because we are just checking validation error.

Reply:
That is the same for all tests that you are only concern with the validation error, and also because we are checking update and store in other tests, so no need to worry about the DB in the validation tests.